components/DetailCountry/index.js

```diff
@@ -0,0 +1,73 @@
+import { useRouter } from "next/router";
+import { styled } from "styled-components";
+
+export default function DetailCountry({ countries }) {
+  const router = useRouter();
+  const currentPage = router.query.detailCountry;
+
+  if (!currentPage) {
+    return null;
+  }
+
+  const currentCountry = countries.find(
+    (country) => country.name === currentPage
+  );
+
+  const { name, startDate, endDate } = currentCountry;
+
+  return (
+    <>
+      <StyledCountries>
+        <StyledCountry>
+          <StyledCountryName>{name}</StyledCountryName>
+          <StyledDiv>
+            <StyledDateHeader>Reisezeitraum:</StyledDateHeader>
+            <StyledDate>{`${startDate} bis ${endDate}`}</StyledDate>
+          </StyledDiv>
+        </StyledCountry>
+      </StyledCountries>
+    </>
+  );
+}
+
+const StyledDiv = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: space-between;
+  gap: 0;
+`;
+
+const StyledCountryName = styled.h3`
+  text-align: center;
+  margin-top: 0;
+`;
+
+const StyledDateHeader = styled.h5`
+  margin: 0 auto;
+`;
+
+const StyledDate = styled.h6`
+  margin: 0 auto;
+`;
```

Comment:
Just as a side note, you don't want to skip any numbers in your headings for SEO reasons. If you have an `h3` element, use an `h4` next. Don't skip forward to `h5`.